From be1703d30565b74228c0809ee043d1b8a3b50b31 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Tim-Philipp=20M=C3=BCller?= <tim@centricular.net>
Date: Sat, 3 Nov 2012 13:51:12 +0000
Subject: [PATCH 1/2] opencv: fix warnings and build against opencv 2.4.x

gstpyramidsegment.c: In function 'gst_pyramid_segment_chain':
gstpyramidsegment.c:307:3: error: implicit declaration of function
'cvPyrSegmentation' [-Werror=implicit-function-declaration]
gstpyramidsegment.c:307:3: error: nested extern declaration of
'cvPyrSegmentation' [-Werror=nested-externs]

https://bugzilla.gnome.org/show_bug.cgi?id=687237
---
 ext/opencv/gstpyramidsegment.h | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/ext/opencv/gstpyramidsegment.h b/ext/opencv/gstpyramidsegment.h
index 5c210ca..47123f4 100644
--- a/ext/opencv/gstpyramidsegment.h
+++ b/ext/opencv/gstpyramidsegment.h
@@ -49,6 +49,10 @@
 #include <gst/gst.h>
 #include <cv.h>
 
+#if CV_MAJOR_VERSION * 100 + CV_MINOR_VERSION >= 204
+#include <opencv2/legacy/legacy.hpp>
+#endif
+
 G_BEGIN_DECLS
 /* #defines don't like whitespacey bits */
 #define GST_TYPE_PYRAMID_SEGMENT \
-- 
1.8.1.4

